test/mod2pi.jl

```diff
@@ -194,3 +194,70 @@ testModPi()
+             3*pi/4-0.1, # -3π/4 <= x <= 3π/4
+             pi-0.1, Float64(pi), # -4π/4 <= x <= 4π/4
+             5*pi/4-0.1, # -5π/4 <= x <= 5π/4
+             6*pi/4-0.1, 6*pi/4, # -6π/4 <= x <= 6π/4
+             7*pi/4-0.1, # -7π/4 <= x <= 7π/4
+             2*pi, 2*pi-0.1, # -8π/4 <= x <= 8π/4
+             9*pi/4-0.1, # -9π/4 <= x <= 9π/4
+             2.0^10*pi/4, # -2.0^20π/2 <= x <= 2.0^20π/2
+             2.0^30*pi/4, # |x| >= 2.0^20π/2, idx < 0
+             2.0^80*pi/4] # |x| >= 2.0^20π/2, idx > 0-0.22370138542135648
+
+     # ieee754_rem_pio2_return contains the returned value from the ieee754_rem_pio2
+     # function in openlibm https://github.com/JuliaLang/openlibm/blob/master/src/e_rem_pio2.c
+     ieee754_rem_pio2_return = [1.5707963267948966 1.5707963267948966;
+                                1.0471975511965979 -1.0471975511965979;
+                                0.10000000000000014 -0.10000000000000014;
+                                6.123233995736766e-17 -6.123233995736766e-17;
+                                -0.6853981633974481 0.6853981633974481;
+                                0.10000000000000021 -0.10000000000000021;
+                                1.2246467991473532e-16 -1.2246467991473532e-16;
+                                -0.6853981633974481 0.6853981633974481;
+                                0.09999999999999983 -0.09999999999999983;
+                                1.8369701987210297e-16 -1.8369701987210297e-16;
+                                -0.6853981633974484 0.6853981633974484;
+                                2.4492935982947064e-16 -2.4492935982947064e-16;
+                                0.0999999999999999 -0.0999999999999999;
+                                -0.6853981633974484 0.6853981633974484;
+                                3.135095805817224e-14 -3.135095805817224e-14;
+                                3.287386219680602e-8 -3.287386219680602e-8;
+                                -0.1757159771004682 0.1757159771004682]'
+
+    for (i, case) in enumerate(cases)
+        # negative argument
+        retmf = sum(Base.Math.rem_pio2_kernel(-case)[2:3])
+        @test sum(retmf) == ieee754_rem_pio2_return[1, i]
+        diff = Float64(mod(big(-case), big(pi)/2))-sum(Base.Math.rem_pio2_kernel(-case)[2:3])
+        @test abs(diff) in (0.0, 1.5707963267948966, 1.5707963267948968)
+        # positive argument
+        ret = Base.Math.rem_pio2_kernel(case)[2:3]
+        @test sum(ret) == ieee754_rem_pio2_return[2, i]
+        diff = Float64(mod(big(case), big(pi)/2))-sum(Base.Math.rem_pio2_kernel(case)[2:3])
+        @test abs(diff) in (0.0, 1.5707963267948966, 1.5707963267948968)
+    end
+end
+@testset "rem_pio2_kernel and mod2pi" begin
+    for int in (3632982096228748, 1135326194816)
+        bignum = int*big(pi)/2+0.00001
+        bigrem = rem(bignum, big(pi)/2, RoundDown)
+        fnum = Float64(bignum)
+        @test mod2pi(fnum) == sum(Base.Math.rem_pio2_kernel(fnum)[2:3])
```

Comment:
The indexing here is triggering a `BoundsError`, since `rem_pio2_kernel` returns a 2-tuple.